cli/cmd/connect.go

```diff
@@ -122,18 +137,53 @@ func isCredentialsURI(str string) bool {
 	// user:password@host:port
 	tcpReStr := `(tcp://)?` + userpassRe + `@([\w\.-]+:\d+)`
 	// unix://user:password@../path
+	// unix://user:password@~/path
 	// unix://user:password@/path
 	// unix://user:password@path
-	unixReStr := `unix://` + userpassRe + `@[./@]*[^\./@]+.*`
-	// user:password@./path
+	unixReStr := `unix://` + userpassRe + `@` + uriPathPrefixRe + `[^\./@]+.*`
+	// user:password@../path
+	// user:password@~/path
 	// user:password@/path
-	pathReStr := userpassRe + `@\.?/[^\./].*`
+	// user:password@./path
+	pathReStr := userpassRe + `@` + systemPathPrefixRe + `[^\./].*`
 
 	uriReStr := "^((" + tcpReStr + ")|(" + unixReStr + ")|(" + pathReStr + "))$"
 	uriRe := regexp.MustCompile(uriReStr)
 	return uriRe.MatchString(str)
 }
 
+// parseBaseURI parses an URI and returns:
+// (network, address)
+func parseBaseURI(URI string) (string, string) {
+	var network, address string
+	URILen := len(URI)
```

Comment:
A variables that starts with a capital letter may be confusing in Go. Up to you.

```suggestion
// parseBaseURI parses an URIi and returns:
// (network, address)
func parseBaseURI(uri string) (string, string) {
	var network, address string
	uriLen := len(uri)
```